python/src/aac/lang/references.py

```diff
@@ -0,0 +1,23 @@
+"""Module for AaC Language functions related to definition references."""
+
+from aac.lang.definitions.definition import Definition
+
+
+def get_definition_references_from_list(
+    source_definition: Definition, definitions_to_search: list[Definition]
+) -> list[Definition]:
+    """
+    Return a subset of Definitions that reference the source_definition.
+
+    Args:
+        source_definition (Definition): The definition that is being referenced
+        definitions_to_search (list[Definition]): The definitions to search through for references
+
+    Return:
+        A list of Definitions that reference the source_definition
+    """
+
+    def _filter_definitions_by_reference(definition_to_filter: Definition) -> bool:
```

Comment:
I wrote out my response in your other comment. Indentation is only applicable to the declaration of the inner function, not when it's called. However, the Python community seems to not care, and given our complexity rules then it should obvious enough that the function being referred to is an inner function 20 or so lines above and not a public module function. When in Rome...

Reply:
1. I don't think the underscore solves the issue of knowing it's an inner function because we also use a leading underscore for top-level functions, so we wouldn't have a way to distinguish that.
2. If we have an inner function separated from where it's called by 20 lines, then I'd argue that function is too big anyways. I realize that I have not commented on function length in the PR, mostly because we haven't established any guidelines for that but I agree with the idea of functions being very small as described in [this](https://vimeo.com/12643301) video (or, if you have the [Clean Code](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882/ref=sr_1_1?crid=B2J17SEP2OMS&keywords=clean+code&qid=1649196868&sprefix=clean+code%2Caps%2C125&sr=8-1) book, it's described in Chapter 3).

Reply:
My bad, I didn't make my last comment clear. I'll capitulate and not underscore my inner functions. 

For clarity, my position on the underscore is that it denotes that the function is defined in the current file regardless of whether it is defined at the top-level of the file/module or inside a function. In my missing comment, I made the point that the underscore before a function immediately denotes a non-public module function, which means that the function has to be defined in the current file which provides some immediate feedback to the developer. 

To your second point, you have to call the inner function after it's defined, so you're already a few lines from the declaration of the function name. Our flake complexity rules will handle this, so I'm not worried about it getting out of control.